--- a/deps/v8/src/sandbox/sandboxed-pointer-inl.h
+++ b/deps/v8/src/sandbox/sandboxed-pointer-inl.h
@@ -31,13 +31,15 @@ V8_INLINE void WriteSandboxedPointerField(Address field_address,
                                           PtrComprCageBase cage_base,
                                           Address pointer) {
 #ifdef V8_ENABLE_SANDBOX
-  // The pointer must point into the sandbox.
-  CHECK(GetProcessWideSandbox()->Contains(pointer));
+  WriteMaybeUnalignedValue<Address>(field_address, pointer);
+
+  // // The pointer must point into the sandbox.
+  // CHECK(GetProcessWideSandbox()->Contains(pointer));

-  Address offset = pointer - cage_base.address();
-  SandboxedPointer_t sandboxed_pointer = offset << kSandboxedPointerShift;
-  base::WriteUnalignedValue<SandboxedPointer_t>(field_address,
-                                                sandboxed_pointer);
+  // Address offset = pointer - cage_base.address();
+  // SandboxedPointer_t sandboxed_pointer = offset << kSandboxedPointerShift;
+  // base::WriteUnalignedValue<SandboxedPointer_t>(field_address,
+  //                                               sandboxed_pointer);
 #else
   WriteMaybeUnalignedValue<Address>(field_address, pointer);
 #endif
 