--- a/vcbuild.bat
+++ b/vcbuild.bat
@@ -68,6 +68,7 @@ set doc=
 set extra_msbuild_args=
 set compile_commands=
 set exit_code=0
+set v8_ptr_compress=
 
 :next-arg
 if "%1"=="" goto args-done
@@ -142,6 +143,7 @@ if /i "%1"=="no-shared-roheap" set no_shared_roheap=1&goto arg-ok
 if /i "%1"=="doc"           set doc=1&goto arg-ok
 if /i "%1"=="binlog"        set extra_msbuild_args=/binaryLogger:out\%config%\node.binlog&goto arg-ok
 if /i "%1"=="compile-commands" set compile_commands=1&goto arg-ok
+if /i "%1"=="v8_ptr_compress"   set v8_ptr_compress=1&goto arg-ok
 
 echo Error: invalid command line option `%1`.
 exit /b 1
@@ -199,6 +201,7 @@ if defined openssl_no_asm   set configure_flags=%configure_flags% --openssl-no-a
 if defined no_shared_roheap set configure_flags=%configure_flags% --disable-shared-readonly-heap
 if defined DEBUG_HELPER     set configure_flags=%configure_flags% --verbose
 if defined compile_commands set configure_flags=%configure_flags% -C
+if defined v8_ptr_compress  set configure_flags=%configure_flags% --experimental-enable-pointer-compression
 if "%target_arch%"=="x86" if "%PROCESSOR_ARCHITECTURE%"=="AMD64" set configure_flags=%configure_flags% --no-cross-compiling
 
 if not exist "%~dp0deps\icu" goto no-depsicu
